### PR TITLE
📖 Update Component config tutorial to use go/v4 plugin

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description

In this PR we are updating the tutorial to be based on the go/v4 plugin and not on go/v3 one. 

Motivation: https://github.com/kubernetes-sigs/kubebuilder/issues/3117

### CODE REVIEW

File: docs/book/src/component-config-tutorial/testdata/project/config/crd/patches/cainjection_in_projectconfigs.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    cert-manager.io/inject-ca-from: $(CERTIFICATE_NAMESPACE)/$(CERTIFICATE_NAME)
+    cert-manager.io/inject-ca-from: CERTIFICATE_NAMESPACE/CERTIFICATE_NAME
```
Comment (Kavinjsir) [inline docs/book/src/component-config-tutorial/testdata/project/config/crd/patches/cainjection_in_projectconfigs.yaml:6]: Just wondering why do we drop `$()` to make it a plain text?
(Assuming the origin one is used to be substituted by kustomize?)

**Review by Kavinjsir — APPROVED** [review]

File: docs/book/src/component-config-tutorial/testdata/project/config/crd/patches/cainjection_in_projectconfigs.yaml (hunk 1 above)
Comment (camilamacedo86) [reply docs/book/src/component-config-tutorial/testdata/project/config/crd/patches/cainjection_in_projectconfigs.yaml:6]: Yes, this change came from kustomize/v2

**Comment by camilamacedo86** [comment]
Would be possible @Kavinjsir lgtm this one for we get this one merged?

**Pull request merged**